pkg/scd/operational_intents_handler.go

```diff
@@ -739,49 +814,15 @@ func (a *Server) upsertOperationalIntentReference(ctx context.Context, authorize
 			}
 
 		} else {
-			// Use existing Subscription
-			sub, err = r.GetSubscription(ctx, validParams.subscriptionID)
+			// Use existing Subscription and adapt it if needed and possible
+			sub, err = dependOnExistingSubscriptionIfAllowedAndCovered(ctx, r, manager, validParams)
 			if err != nil {
-				return stacktrace.Propagate(err, "Unable to get Subscription")
-			}
-			if sub == nil {
-				return stacktrace.NewErrorWithCode(dsserr.BadRequest, "Specified Subscription %s does not exist", validParams.subscriptionID)
-			}
-			if sub.Manager != manager {
-				return stacktrace.Propagate(
-					stacktrace.NewErrorWithCode(dsserr.PermissionDenied, "Specificed Subscription is owned by different client"),
-					"Subscription %s owned by %s, but %s attempted to use it for an OperationalIntent", validParams.subscriptionID, sub.Manager, manager)
-			}
-			updateSub := false
-			if sub.StartTime != nil && sub.StartTime.After(*validParams.uExtent.StartTime) {
-				if sub.ImplicitSubscription {
-					sub.StartTime = validParams.uExtent.StartTime
-					updateSub = true
-				} else {
-					return stacktrace.NewErrorWithCode(dsserr.BadRequest, "Subscription does not begin until after the OperationalIntent starts")
-				}
-			}
-			if sub.EndTime != nil && sub.EndTime.Before(*validParams.uExtent.EndTime) {
-				if sub.ImplicitSubscription {
-					sub.EndTime = validParams.uExtent.EndTime
-					updateSub = true
-				} else {
-					return stacktrace.NewErrorWithCode(dsserr.BadRequest, "Subscription ends before the OperationalIntent ends")
-				}
-			}
-			if !sub.Cells.Contains(validParams.cells) {
-				if sub.ImplicitSubscription {
-					sub.Cells = s2.CellUnionFromUnion(sub.Cells, validParams.cells)
-					updateSub = true
-				} else {
-					return stacktrace.NewErrorWithCode(dsserr.BadRequest, "Subscription does not cover entire spatial area of the OperationalIntent")
-				}
-			}
-			if updateSub {
-				sub, err = r.UpsertSubscription(ctx, sub)
-				if err != nil {
-					return stacktrace.Propagate(err, "Failed to update existing Subscription")
-				}
+				// Propagating the code here is important for properly handling all failure cases in the caller
```

Comment:
Doesn't the simple `Propagate` does that in any case? I thought the error code would be unwrapped at the top of the stack.

Reply:
You're right: if the passed cause has a code, it will be re-used.

(But, for the record, if it has none, there will be no attempt to go look a level deeper for a code)